server/leafnode.go

```diff
@@ -2164,8 +2175,44 @@ func (s *Server) leafNodeFinishConnectProcess(c *client) {
 	// Spin up the write loop.
 	s.startGoRoutine(func() { c.writeLoop() })
 
+	cid := c.cid
+	c.mu.Unlock()
 	c.Debugf("Remote leafnode connect msg sent")
 
+	// timeout leafNodeFinishConnectProcess
+	time.AfterFunc(s.getOpts().PingInterval, func() {
```

Comment:
The PingInterval can be quite high (like 2min by default). Should we have a smaller interval instead (as a constant for now, not exposed)?

Reply:
A constant of 2 seconds? Or AuthTimeout?

Reply:
Maybe AuthTimeout but 2sec at minimum?

Reply:
went with 2 seconds

Reply:
fixed